src/lib.rs

```diff
@@ -1738,6 +1741,13 @@ ffi!(fn pgp_export_keydata(session: *mut Session,
 
     let mut keydata = Vec::new();
     if secret {
+        // No-op if there is no passphrase (NULL).
+        // If there is a passphrase, only encrypted parts are decrypted.
+        let cert = decrypt_cert(cert, passphrase)?;
+
+        // No-op if there is no passphrase (NULL).
+        let cert = encrypt_cert(cert, passphrase)?;
+
```

Comment:
This looks circular? This would make sense if the passphrases are different (like in `manage_passphrases`). But since the passphrases are the same, why is this needed?

Reply:
I want to both support the case "not encrypted -> encrypted" and "encrypted -> encrypted (both with the same passphrase)" explicitly, and there is only one passphrase per account in our setup in this case.

Reply:
Ah, make sense. Can you comment this at the top?

```
// (Re-)encrypt with the passphrase. This allows both cases to work:
// 1. not encrypted -> encrypted
// 2. encrypted -> encrypted (this is effectively a no-op)
```